app/presenters/finder_presenter.rb

```diff
@@ -145,57 +141,8 @@ def text_metadata_keys
     metadata.select { |f| f.type == "text" }.map(&:key)
   end
 
-  def default_sort_option
-    sort
-      &.detect { |option| option['default'] }
-  end
-
-  def default_sort_option_value
-    default_sort_option
-      &.dig('name')
-      &.parameterize
-  end
-
-  def default_sort_option_key
-    default_sort_option
-      &.dig('key')
-  end
-
-  def relevance_sort_option
-    sort
-      &.detect { |option| %w(relevance -relevance).include?(option['key']) }
-  end
-
-  def relevance_sort_option_value
-    relevance_sort_option
-      &.dig('name')
-      &.parameterize
-  end
-
   def sort_options
-    return [] if sort.blank?
-
-    options = sort.collect do |option|
-      [
-        option['name'],
-        option['name'].parameterize,
-        {
-          'data-track-category' => 'dropDownClicked',
-          'data-track-action' => 'clicked',
-          'data-track-label' => option['name']
-        }
-      ]
-    end
-
-    disabled_option = keywords.blank? ? relevance_sort_option_value : ''
-
-    selected_option = if values['order'].present? && sort.any? { |option| option['name'].parameterize == values['order'] }
-                        values['order']
-                      else
-                        default_sort_option_value
-                      end
-
-    options_for_select(options, selected: selected_option, disabled: disabled_option)
+    @sort_options ||= SortPresenter.new(content_item, values)
```

Comment:
I'm wondering if it is maybe better to instantiate this class outside of this class and move ```atom_feed_enabled?``` etc into ```SortPresenter```

Reply:
It might simplify the code a bit as well as there are quite a number of references to ```finder. sort_options.method``` throughout the code

Reply:
This methods is not related to presenting sort so I’d argue that it could stay here / be moved elsewhere. Looking at sort presenter to check whether one can render the atom feed might look a bit odd.

It’s a shame things rely on sort for their logic. Perhaps we could rewrite this a bit so they don’t?